wasm/jit/jit_amd64.go

```diff
@@ -1440,6 +1456,192 @@ func (c *amd64Compiler) emitSimpleBinaryOp(instruction obj.As) error {
+	target := c.locationStack.peek() // Note this is peek!
+	if err := c.ensureOnGeneralPurposeRegister(target); err != nil {
+		return err
+	}
+
+	tmpReg, err := c.allocateRegister(generalPurposeRegisterTypeFloat)
+	if err != nil {
+		return err
+	}
+
+	// First we move the sign-bit mask (placed in memory) to the tmp register,
+	// since we cannot take XOR directly with float reg and const.
+	movToTmp := c.newProg()
+	var maskAddr uintptr
+	if o.Type == wazeroir.Float32 {
+		movToTmp.As = x86.AMOVL
+		maskAddr = float32SignBitMaskAddress
+	} else {
+		movToTmp.As = x86.AMOVQ
+		maskAddr = float64SignBitMaskAddress
+	}
+	movToTmp.From.Type = obj.TYPE_MEM
+	movToTmp.From.Offset = int64(maskAddr)
+	movToTmp.To.Type = obj.TYPE_REG
+	movToTmp.To.Reg = tmpReg
+	c.addInstruction(movToTmp)
+
+	// Negate the value by XOR it with the sign-bit mask.
+	negate := c.newProg()
+	if o.Type == wazeroir.Float32 {
+		negate.As = x86.AXORPS
+	} else {
+		negate.As = x86.AXORPD
+	}
+	negate.From.Type = obj.TYPE_REG
+	negate.From.Reg = tmpReg
+	negate.To.Type = obj.TYPE_REG
+	negate.To.Reg = target.register
+	c.addInstruction(negate)
+	return nil
+}
+
+// compileCeil emits the instructions to take the ceil of the top value of float type on the stack.
+// For example, stack [..., 1.123] results in [..., 2.0]. This is equivalent to "math.Ceil".
+func (c *amd64Compiler) compileCeil(o *wazeroir.OperationCeil) (err error) {
+	// Internally, ceil can be performed via ROUND instruction with 0x02 mode.
+	// See https://android.googlesource.com/platform/bionic/+/882b8af/libm/x86_64/ceilf.S for example.
+	return c.emitRoundInstruction(o.Type == wazeroir.Float32, 0x02)
+}
+
+// compileFloor emits the instructions to take the floor of the top value of float type on the stack.
+// For example, stack [..., 1.123] results in [..., 1.0]. This is equivalent to "math.Floor".
+func (c *amd64Compiler) compileFloor(o *wazeroir.OperationFloor) (err error) {
+	// Internally, floor can be performed via ROUND instruction with 0x01 mode.
+	// See https://android.googlesource.com/platform/bionic/+/882b8af/libm/x86_64/floorf.S for example.
+	return c.emitRoundInstruction(o.Type == wazeroir.Float32, 0x01)
+}
+
+// compileTrunc emits the instructions to take the integer value of the top value of float type on the stack.
+// For example, stack [..., 1.9] results in [..., 1.0]. This is equivalent to "math.Trunc".
+func (c *amd64Compiler) compileTrunc(o *wazeroir.OperationTrunc) error {
+	// Internally, trunc can be performed via ROUND instruction with 0x03 mode.
+	// See https://android.googlesource.com/platform/bionic/+/882b8af/libm/x86_64/truncf.S for example.
+	return c.emitRoundInstruction(o.Type == wazeroir.Float32, 0x03)
+}
+
+// compileNearest emits the instructions to take the nearest integer value of the top value of float type on the stack.
+// For example, stack [..., 1.9] results in [..., 2.0]. This is NOT equivalent to "math.Round" and instead has the same
```

Comment:
If not equiv to math.round, is there a better example we can find which doesn't result in the same value?

Reply:
https://github.com/tetratelabs/wazero/pull/119/commits/ee4aadab3e6f1faffe2cda117ec7e5ec17876bee added the example and also the test case for it!